core/state-machine/src/backend.rs

```diff
@@ -118,25 +140,30 @@ pub trait Backend<H: Hasher>: std::fmt::Debug {
 	}
 
 	/// Try convert into trie backend.
-	fn as_trie_backend(&mut self) -> Option<&TrieBackend<Self::TrieBackendStorage, H>> {
+	fn as_state_backend(&mut self) -> Option<
+		&StateBackend<Self::TrieBackendStorage, H, Self::KvBackend>
+	> {
 		None
 	}
 
 	/// Calculate the storage root, with given delta over what is already stored
 	/// in the backend, and produce a "transaction" that can be used to commit.
 	/// Does include child storage updates.
-	fn full_storage_root<I1, I2i, I2>(
+	fn full_storage_root<I1, I2i, I2, I3>(
```

Comment:
Yes kv does not influe in the root, it is mainly here to complete the transaction.
Ideally we could have a function that does only the root: this would be good from my point of view (there is quite a few place where we do not use the produced transaction).